src/main/tools/build-runfiles-windows.cc

```diff
@@ -257,25 +257,31 @@ class RunfilesCreator {
     wstring dummy_target = runfiles_base_dir + L"\\dummy_target";
 
     // Try creating symlink with admin privilege
-    if (CreateSymbolicLinkW(dummy_link.c_str(), dummy_target.c_str(), 0)) {
-      DeleteFileOrDie(dummy_link);
-      return true;
-    }
-
-    // If we couldn't create symlink, print out an error message and exit.
-    if (GetLastError() == ERROR_PRIVILEGE_NOT_HELD) {
-      die(L"CreateSymbolicLinkW failed:\n%hs\n",
-          "Bazel needs to create symlink for building runfiles tree.\n"
-          "Creating symlink on Windows requires either of the following:\n"
-          "    1. Program is running with elevated privileges (Admin rights).\n"
-          "    2. The system version is Windows 10 Creators Update (1703) or "
-          "later and "
-          "developer mode is enabled.",
-          GetLastErrorString().c_str());
-    } else {
-      die(L"CreateSymbolicLinkW failed: %hs", GetLastErrorString().c_str());
+    if (!CreateSymbolicLinkW(dummy_link.c_str(), dummy_target.c_str(), 0)) {
+      // on a rare occasion the dummy_link may exist from a previous run
+      if (GetLastError() == ERROR_ALREADY_EXISTS) {
+        wstring target, werror;
```

Comment:
I think the problem happens when we run the same binary multiple times at once. All runs will try to create and delete this dummy_link file. This approach could still fail if another run deleted the `dummy_link` between receiving `ERROR_ALREADY_EXISTS` and `ReadSymlink`. Is there any better way to avoid the race condition? Like adding a random suffix (maybe the [pid](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentprocessid)?)

Reply:
Interesting idea, but why would bazel run the process multiple times? In my experience the dummy_link is actually left behind in the runfiles root. Adding the pid (or any random number) will greatly decrease the risk of encountering such a collision.

Reply:
Oh sorry, I think you are right. Bazel supposes to only run this tool under each runfiles root once in each build. Then this isn't caused by race condition.

Maybe the file got left behind because Bazel somehow exited and terminated the process of build-runfiles-windows before it can delete the dummy_link file? In that case, we can just try to delete the dummy_link file before creating it. Because the logic in this PR actually checks the file created by last run, which might have admin right, but the current run may not have the admin right. WDYT?

Reply:
A yes that's another scenario I did not think of. Ill update the PR to reflect your suggestion.

Reply:
Thanks!